scripts/run_custom_eslint_tests_test.py

```diff
@@ -0,0 +1,132 @@
+        filepath = 'scripts/linters/custom_eslint_checks/rules/'
+        self.proc_args = [node_path, nyc_path, mocha_path, filepath]
+
+        self.cmd_token_list: list[list[str]] = []
+
+        self.sys_exit_code: int = 0
+        def mock_sys_exit(err_code: int) -> None:
+            self.sys_exit_code = err_code
+        self.swap_sys_exit = self.swap(sys, 'exit', mock_sys_exit)
+
+    def test_custom_eslint_tests_failed_due_to_internal_error(self) -> None:
+        class MockTask:
+            def communicate(self) -> tuple[bytes, bytes]:   # pylint: disable=missing-docstring
+                return (
+                    b'All files | 100 | 100 | 100 | 100 | ',
+                    b'Path not found.')
+
+        def mock_popen(
+            cmd_tokens: list[str], **unsued_kwargs: Any) -> MockTask:  # pylint: disable=unused-argument
+            self.cmd_token_list.append(cmd_tokens)
+            return MockTask()
+        swap_popen = self.swap(subprocess, 'Popen', mock_popen)
+
+        with swap_popen, self.print_swap, self.swap_sys_exit:
+            run_custom_eslint_tests.main()
+
+        self.assertIn(self.proc_args, self.cmd_token_list)
+        self.assertIn('Path not found.', self.print_arr)
+        self.assertEqual(1, self.sys_exit_code)
+
+    def test_custom_eslint_tests_failed(self) -> None:
+        class MockTask:
+            def communicate(self) -> tuple[bytes, bytes]:   # pylint: disable=missing-docstring
+                return (
+                    b'1 in 125 tests failing.\n' +
+                    b'All files | 100 | 100 | 100 | 100 | ', b'')
+
+        def mock_popen(
+            cmd_tokens: list[str], **unsued_kwargs: Any) -> MockTask:  # pylint: disable=unused-argument
+            self.cmd_token_list.append(cmd_tokens)
+            return MockTask()
+        swap_popen = self.swap(subprocess, 'Popen', mock_popen)
+
+        with swap_popen, self.print_swap, self.swap_sys_exit:
+            run_custom_eslint_tests.main()
+
+        self.assertIn(self.proc_args, self.cmd_token_list)
+        self.assertIn('Tests not passed', self.print_arr)
+        self.assertEqual(1, self.sys_exit_code)
+
+    def test_custom_eslint_tests_passed(self) -> None:
+        class MockTask:
+            def communicate(self) -> tuple[bytes, bytes]:   # pylint: disable=missing-docstring
+                return (
+                    b'All tests passed\n' +
+                    b'All files | 100 | 100 | 100 | 100 | ', b'')
+
+        def mock_popen(
+            cmd_tokens: list[str], **unsued_kwargs: Any) -> MockTask:  # pylint: disable=unused-argument
+            self.cmd_token_list.append(cmd_tokens)
+            return MockTask()
+        swap_popen = self.swap(subprocess, 'Popen', mock_popen)
+
+        with swap_popen, self.print_swap, self.swap_sys_exit:
+            run_custom_eslint_tests.main()
+
+        self.assertIn(self.proc_args, self.cmd_token_list)
+        self.assertIn('All tests passed', self.print_arr)
```

Comment:
Can we check the exit code (in this case probably that `sys.exit` was not called)?

Reply:
Done!